spirv_msl.cpp

```diff
@@ -1394,9 +1458,17 @@ string CompilerMSL::to_qualified_member_name(const SPIRType &type, uint32_t inde
 string CompilerMSL::ensure_valid_name(string name, string pfx)
 {
 	if (name.size() >= 2 && name[0] == '_' && isdigit(name[1]))
+	{
 		return join(pfx, name);
+	}
+	else if (std::find(reserved_names.begin(), reserved_names.end(), name) != reserved_names.end())
```

Comment:
Can we handle this in the same way as `func_name_overrides`?
	- Make the collection a component of the instance so that it can be modified by any possible subclasses or an API (unless we know the list is fixed and we've caught them all).
	- Append `0` instead of prepending `m`. For consistency...and the `0` makes it more obvious that it is a system-generated mod (the prepended `m` for numeric-only members is because they are members...I guess we can rename `ensure_valid_name` to make this clearer).